.github/workflows/build.yml

```diff
@@ -0,0 +1,119 @@
+name: Publish and Deploy
+run-name: Publish and Deploy (PDFs) on PR Merge
+
+# Definizione dei permessi
+permissions:
+  contents: write
+  pages: write
+  id-token: write
+
+# Trigger: pull request sul ramo base
+on:
+  pull_request:
+    branches: 
+      - main
+    types: [closed]
+
+# Unità di lavoro all'interno del workflow
+jobs:
+  docker:
+  
+    # Controllo se la pull request è stata unita al ramo base
+    if: github.event.pull_request.merged == true
+    
+    # Esecuzione in ambiente Ubuntu con le ultime versioni di pacchetti e librerie
+    runs-on: ubuntu-latest
+    
+    # Azioni atomiche eseguite in sequenza all'interno del job
+    steps:
+    
+    # Clonazione del repository sorgente all'interno dell'ambiente di esecuzione del job
+    - name: Checkout Source Repository
+      uses: actions/checkout@v4
+      with:
+          # Recupero tutta la cronologia (branch e tag)
+          fetch-depth: 0
+
+    # Salvataggio e ripristino della cache
+    - name: Cache Docker
+      uses: actions/cache@v4
+      id: cache-docker
+      with:
+        path: ${{ github.workspace }}/tools/docker
+        key: ${{ runner.os }}-docker-${{ hashFiles('**/tools/docker-compose.yml') }}-${{ hashFiles('**/tools/docker/Dockerfile') }}
+        restore-keys: |
+          ${{ runner.os }}-docker
+
+    # Creazione e avvio del contenitore Docker
+    - name: Start Docker Container
+      run: |
+       cd ${{ github.workspace }}/tools
+       docker compose up -d
+
+    # Compilazione dei documenti tramite shell interattiva all'interno del contenitore Docker
+    - name: Compile Documents
+      run: |
+        cd ${{ github.workspace }}/tools
+        docker exec argo-docs-latexpdfbuilder-1 bash -c 'cd /data && chmod +x *.sh && ./compilescript.sh -d 2_RTB'
+
```

Comment:
Può aver senso l'aggiunta di blocchi try-catch o controlli più dettagliati per gestire eventuali errori durante l'esecuzione delle azioni?

name: Compile Documents
  run: |
    # Esegui la compilazione dei documenti
    # Gestisci eventuali errori qui
    # Esempio di controllo del codice di stato dell'azione precedente
    if [ $? -ne 0 ]; then
      echo "Errore durante la compilazione dei documenti."
      exit 1
    fi

Reply:
Avevo fatto dei test al riguardo. Se la compilazione dei documenti fallisce, il workflow continua, semplicemente quando cerca la tabella 2_RTB non troverà niente dentro.

Reply:
La compilazione dei documenti viene fatta in una shell interattiva, fare controlli quando si esce dalla shell è un po' complesso, meglio proseguire come se non fosse successo niente. Genero un artefatto proprio per questo motivo, in modo tale da poter controllare quali documenti sono stati compilati e quali no